geonode_mapstore_client/client/js/components/FiltersForm/FilterItems.jsx

```diff
@@ -100,15 +106,16 @@ function FilterItems({
                             const active = customFilters.find(value => value === item.id);
                             return (
                                 <Checkbox
+                                    key={item.id}
                                     className="gn-sub-filter-items"
                                     type="checkbox"
                                     checked={!!active}
                                     value={item.id}
                                     onChange={() => {
                                         onChange({
-                                            f: active
-                                                ? customFilters.filter(value => value !== item.id)
-                                                : [...customFilters, item.id]
+                                            f: !!active
+                                                ? setActiveChildFilters(customFilters, item)
+                                                : setChildFilters(customFilters, item)
```

Comment:
I think we don't need a parentId property because the field.id is already the parent id neither the utils.
Maybe I missing something but could you verify if this is enough?

```js
const filterChild = () => {
    return field.items && field.items.map((item) => {
        const active = customFilters.find(value => value === item.id);
        return (
            <Checkbox
                key={item.id}
                className="gn-sub-filter-items"
                type="checkbox"
                checked={!!active}
                value={item.id}
                onChange={() => {
                    onChange({
                        f: active
                            ? customFilters.filter(value => item.id !== value)
                            : [...customFilters, item.id, field.id]
                    });
                }}
            >
                <Message msgId={item.labelId}/>
            </Checkbox>
        );
    } );
};
```